src/components/documents/DocumentListItem.js

```diff
@@ -1,27 +1,81 @@
-import React, { Component } from 'react';
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
+import DeleteIcon from '@material-ui/icons/Delete';
+import IconButton from '@material-ui/core/IconButton';
 import ListItem from '@material-ui/core/ListItem';
 import ListItemText from '@material-ui/core/ListItemText';
+import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
 
-import store from '../../js/store/store';
-import { loadDocument } from '../../js/actions/document-actions';
+import { deleteDocument, loadDocument } from '../../js/actions/document-actions';
 
 const moment = require('moment');
 
 class DocumentListItem extends Component {
+  state = {
+    open: false,
+  };
+
+  handleClickOpen = () => {
+    this.setState({ open: true });
+  };
+
+  handleClose = () => {
+    this.setState({ open: false });
+  };
+
   selectDocument = () => {
-    // i feel its cheating (need go for mapdispatchprops.....). Or not :/
-    store.dispatch(loadDocument(this.props.id));
+    this.props.selectDocument(this.props.id);
+  }
+
+  deleteDocument = () => {
+    this.handleClose();
+    this.props.deleteDocument(this.props.id);
   }
 
   render() {
     return (
-      <ListItem button divider onClick={this.selectDocument}>
-        <ListItemText
-          primary={this.props.title}
-          secondary={`${moment(this.props.created).format('YYYY-MM-DD')} (${moment(this.props.created).fromNow()})`}
-        />
-      </ListItem>
+      <Fragment >
+        <Dialog
```

Comment:
You should not duplicate this dialog for every item. It should be one for all in parent component.